eventbrite-consents/src/main/scala/com/gu/identity/eventbriteconsents/clients/IdentityClient.scala

```diff
@@ -16,6 +20,27 @@ class IdentityClient(idapiUrl: String, idapiAccessToken: String) {
       .postData(IdapiConsentUpdate(emailAddress, Vector("events")).asJson.noSpaces)
       .asString
 
-    if (result.code != 200) throw new RuntimeException(s"Unexpected response from idapi when syncing email $emailAddress ${result.code} ${result.body}")
+    if (result.code != 200) {
+      handleErrorResponse(result.code, result.body, emailAddress)
+    }
+  }
+
+  def handleErrorResponse(code: Int, body: String, emailAddress: String): Unit = {
+    parse(body).flatMap(_.as[IdapiErrorResponse]) match {
+      case Right(errorResponse: IdapiErrorResponse) => {
+        for {
+          error <- errorResponse.errors
+        } yield if (error.description.contains("Bad email format")) {
+          logger.error("Invalid email address, could not process consents: " + emailAddress)
+        } else {
+          throwRuntimeException
+        }
+      }
+      case _ => throwRuntimeException
+    }
+
+    def throwRuntimeException: Nothing = {
```

Comment:
for functions with side effects it's idiomatic to use `()`, e.g. `throwRuntimeException()`

Reply:
Ah, good point. I'll change that